projects/rocfft/shared/data_gen_device.h

```diff
@@ -65,6 +65,65 @@ __device__ static void set_imag_zero(const size_t pos, Tfloat* xreal, Tfloat* xi
     ximag[pos] = 0.0;
 }
 
+// get grid dimensions for hermitian symmetrizer kernel
+static dim3 generate_hermitian_gridDim(const std::vector<size_t>& length,
+                                       const size_t               batch,
+                                       const size_t               blockSize)
+{
+    dim3 gridDim;
+
+    if(length.size() > 1 && length[0] == 0)
+        throw std::runtime_error("Invalid zero length for impose_hermitian_symmetry");
+
+    if(length.size() > 2 && length[1] == 0)
```

Comment:
```suggestion
    if(!length.empty() && std::count(length.begin(), length.end()-1, static_cast<size_t>(0)) > 0)
```
\+ while we're at it, shall we validate that `batch > 0` as well?